hedera-node/docs/proposals/block-stream-bucket-uploader.md

```diff
@@ -0,0 +1,398 @@
+                "provider": "aws",
+                "endpoint": "https://s3.amazonaws.com",
+                "region": "us-east-1",
+                "bucketName": "hedera-mainnet-blocks"
+            },
+            {
+                "name": "default-gcp-bucket",
+                "provider": "gcp",
+                "endpoint": "https://storage.googleapis.com",
+                "region": "",
+                "bucketName": "hedera-mainnet-blocks"
+            }
+        ]
+        """) @NetworkProperty List<BucketNetworkConfig> buckets,
+    // Updated Node Property for writerModes
+    @ConfigProperty(defaultValue = "[FILE, BUCKETS]") @NodeProperty List<BlockStreamWriterMode> writerModes,
+    // ... existing code ...
+) {}
+
+public record BucketNetworkConfig(
+    @ConfigProperty String name,
+    @ConfigProperty String provider,  // "aws" or "gcp"
+    @ConfigProperty String endpoint,
+    @ConfigProperty String region,    // required for AWS only
+    @ConfigProperty String bucketName
+) {
+    public BucketNetworkConfig {
+        Objects.requireNonNull(name, "Bucket name cannot be null");
+        Objects.requireNonNull(provider, "Provider cannot be null");
+        Objects.requireNonNull(endpoint, "Endpoint cannot be null");
+        Objects.requireNonNull(bucketName, "Bucket name cannot be null");
+        
+        if ("aws".equals(provider)) {
+            Objects.requireNonNull(region, "Region is required for AWS buckets");
+        }
+    }
+}
+```
+
+#### Credentials Configuration
+The credentials are stored in a separate JSON file on disk for security:
+
+```json
+{
+  "credentials": {
+    "default-aws-bucket": {
+      "accessKey": "YOUR_AWS_ACCESS_KEY",
+      "secretKey": "YOUR_AWS_SECRET_KEY"
+    },
+    "default-gcp-bucket": {
+      "accessKey": "YOUR_GCP_ACCESS_KEY",
+      "secretKey": "YOUR_GCP_SECRET_KEY"
+    }
+  }
+}
+```
+
+#### Credentials Records
+```java
+public record BucketCredentials(
+    String accessKey,
+    String secretKey
+) {
+    public BucketCredentials {
+        Objects.requireNonNull(accessKey, "Access key cannot be null");
+        Objects.requireNonNull(secretKey, "Secret key cannot be null");
+    }
+}
+
+public record BucketCredentialsConfig(
+    Map<String, BucketCredentials> credentials
+) {
+    public BucketCredentialsConfig {
+        Objects.requireNonNull(credentials, "Credentials map cannot be null");
+    }
+}
+```
+
+#### Configuration Loading and Updates
+The configuration system needs to handle both network properties and local credentials in a secure and maintainable way.
+
+**BucketConfigurationManager Requirements**
+- Singleton class managing both network properties and local credentials
+- Load credentials from the JSON file specified in the configuration
+- Provide methods to get complete bucket configurations by combining network properties with credentials
+- Handle credential file reload if needed
+- Log appropriate warnings/errors when credentials are missing or invalid
```

Comment:
Should this be a critical error that stops the CN from starting up successfully until we can load the cloud credentials correctly?